### ISSUE TITLE
Questions about the project

### ISSUE THREAD

**Issue by sezaru** [bh1xuw/rust-rocks#18]
First of all, thanks for the library!

I'm developing a rocksdb wrapper for Elixir language.

Currently, I'm using rust-rocksdb library instead of directly wrapping into C/C++ (for safety reasons). I recently found out about your library and became interested in maybe replace rust-rocksdb with it for my wrapper.

I just have some questions that I hope you can answer if you don't mind.

1) Is this library a long-term project? I saw that you already have more functionality implemented compared to rust-rocksdb and also I find your organization way better (ex. You separate database options from column family options) but I was wondering if you pretend to continue develop this project in the future;

2) One issue that I have with rust-rocksdb and I would probably have with your library too is lifetime objects.

For example, you need to define a lifetime to a `Iterator` struct, but if I want to return that iterator pointer via my wrapper (so I can use the iterator in Elixir code) it is not possible because of the lifetime.

As far as I understand, the only reason for the lifetime in `Iterator` is because of the `PhantomData` field: 
`_marker: PhantomData<&'a ()>`.

Is it possible to provide an alternative to this without this field so Rust will not complain when I try to return it? Or maybe you know some other way around it that I'm not aware.

Thanks you for your time!

**Comment by bh1xuw** [comment]
> Is this library a long-term project? 

yes, the linked version will be kept the same as what is avaliable on HomeBrew.

> lifetime of iterator

The lifetime marker is used to guarantee DB lives longer than Iterator.
You can use an unsafe pointer to hold it.

**Comment by sezaru** [comment]
Hey, thanks for the reply.

Another question, do you want PR with new features added? For example, I created a fork to add support for DBWithTtl.

**Comment by bh1xuw** [comment]
:) PRs are welcomed
